src/transformStateWithClones.js

```diff
@@ -6,8 +6,33 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let modifiedState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        modifiedState = { ...modifiedState, ...action.extraData };
+        result.push({ ...modifiedState });
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete modifiedState[key];
+        }
+        result.push(modifiedState);
```

Comment:
Here, you are pushing the `modifiedState` object directly into the `result` array. This can lead to potential issues if `modifiedState` is modified afterwards. It's safer to push a copy of `modifiedState` using the spread operator.

Reply:
It will be resolved after implementing the previous comment